native/src/utils/renderHtml.ts

```diff
@@ -69,39 +84,123 @@ const renderJS = (cacheDictionary: ParsedCacheDictionaryType, allowedIframeSourc
+    window.addEventListener('load', adjustHeight)
+    window.addEventListener('resize', adjustHeight)
+    const details = document.querySelectorAll('details')
+    details.forEach(detail => detail.addEventListener('toggle', adjustHeight))
   })();
 
-  (function handleIframes() {
-    const iframes = document.querySelectorAll('iframe')
-    const allowedIframeSources = ${JSON.stringify(allowedIframeSources)}
+  (function handleIframes () {
+    function capitalizeFirstLetter (word) {
+      return word.charAt(0).toUpperCase() + word.slice(1)
+    }
+    
+    function showMessage(text, element, className) {
+      const textNode = document.createTextNode(text);
+      if(className){
+         element.classList.add(className)
+      }
+      element.appendChild(textNode)
+    }
+    
+    function showSettingsButton(element) {
+      function onClickHandler() {
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${SETTINGS_MESSAGE_TYPE}', openSettings: true }))
+      }
+      const buttonLabel = ${JSON.stringify(t('layout:settings'))}
+      const button = document.createElement('button');
+      button.name = "opt-in-settings-button";
+      button.innerHTML = buttonLabel;
+      button.id = "opt-in-settings-button";
+      button.onclick = onClickHandler;
+      element.appendChild(button);
+    }
+    
+    function showMessageWithSettings(text, iframeContainer) {
+      showMessage(text, iframeContainer, "iframe-info-text")
+      showSettingsButton(iframeContainer)
+      
+    }
+    
+    function showOptIn(text, iframeContainer, source) {
+      function onClickHandler() {
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${IFRAME_MESSAGE_TYPE}', allowedSource: {type: source, allowed: true} }))
+      }
+      const checkbox = document.createElement('input');
+      checkbox.type = "checkbox";
+      checkbox.name = "opt-in-checkbox";
+      checkbox.id = "opt-in-checkbox";
+      checkbox.onclick = onClickHandler;
+      const label = document.createElement('label')
+      label.htmlFor = "opt-in-checkbox";
+      label.appendChild(document.createTextNode(text));
+      iframeContainer.appendChild(label);
+      iframeContainer.appendChild(checkbox);
+    }
+
+    function showBlockMessageWithSettings(text, iframeContainer) {
+      showMessage(text, iframeContainer)
+      showSettingsButton(iframeContainer)
+    }
 
-    iframes.forEach((el) => {
-      if (allowedIframeSources.some(url => el.src.indexOf(url) > 0)) {
-        const url = new URL(el.src)
-        // Add do not track parameter
+    function handleWhiteListedIframeSources (iframe, allowedExternalSourcePermissions, iframeSource, iframeContainer) {
+        // Add do not track parameter (only working for vimeo)
+     if (externalSourcePermissions.some(source => source.type === iframeSource && source.allowed)) {
+        const url = new URL(iframe.src)
         url.searchParams.append('dnt', '1')
```

Comment:
As discussed in the other PR, I think an actual check here that the URL is from vimeo is a good idea.

Reply:
well i find it ugly to hardcode the check and its actually no harm to add it in any case. Since this is a iframe the user won't recognize this parameter

Reply:
But you don't find it ugly to hardcode the parameter? I guess it is no harm done but I don't like the idea of unnecessarily adding a parameter to every (future) video

Reply:
added check